sssom/util.py

```diff
@@ -792,18 +802,43 @@ def get_dict_from_mapping(map_obj: Union[Any, Dict[Any, Any], SSSOM_Mapping]) ->
     :param map_obj: Mapping object
     :return: Dictionary
     """
+    # TODO: Change logic
     map_dict = {}
     for property in map_obj:
-        if isinstance(map_obj[property], list):
-            map_dict[property] = "|".join(
-                enum_value.code.text
-                for enum_value in map_obj[property]
-                if type(enum_value).__name__ == MatchTypeEnum._defn.name
-            )
-        elif type(map_obj[property]).__name__ == PredicateModifierEnum._defn.name:
-            map_dict[property] = map_obj[property].code.text
+        if map_obj[property] is not None:
+            if isinstance(map_obj[property], list):
+                # IF object is an enum
+                if (
+                    SCHEMA_DICT["slots"][property]["range"]
+                    in SCHEMA_DICT["enums"].keys()
+                ):
+                    # IF object is a multivalued enum
+                    if SCHEMA_DICT["slots"][property]["multivalued"]:
+                        map_dict[property] = "|".join(
+                            enum_value.code.text for enum_value in map_obj[property]
+                        )
+                    # If object is NOT multivalued BUT an enum.
+                    else:
+                        map_dict[property] = map_obj[property].code.text
+                # IF object is NOT an enum but a list
+                else:
+                    map_dict[property] = "|".join(
+                        enum_value for enum_value in map_obj[property]
+                    )
+            # IF object NOT a list
+            else:
+                # IF object is an enum
+                if (
+                    SCHEMA_DICT["slots"][property]["range"]
+                    in SCHEMA_DICT["enums"].keys()
+                ):
+                    map_dict[property] = map_obj[property].code.text
+                else:
+                    map_dict[property] = map_obj[property]
         else:
-            map_dict[property] = map_obj[property]
+            # IF map_obj[property] is None:
+            map_dict[property] = ""
```

Comment:
Is this ok, even if its `double`? Probably, just double checking.

Reply:
Good catch! fixed it!